src/saml.ts

```diff
@@ -1022,12 +1022,16 @@ class SAML {
           const _confirmData = _subjectConfirmation.SubjectConfirmationData?.[0];
           if (_confirmData?.$) {
             const subjectNotBefore = _confirmData.$.NotBefore;
+
             const subjectNotOnOrAfter = _confirmData.$.NotOnOrAfter;
-            const maxTimeLimitMs = this.calcMaxAgeAssertionTime(
-              this.options.maxAssertionAgeMs,
-              subjectNotOnOrAfter,
-              assertion.$.IssueInstant
-            );
+
+            const maxTimeLimitMs = subjectNotOnOrAfter
```

Comment:
Correction to this logic should be in `this.calcMaxAgeAssertionTime()`, not here. We should be able to handle bad data coming in and still return a proper value per the existing logic.

Reply:
@cjbarth, `checkTimestampsValidityError` properly handles `maxTimeLimitMs` that is `undefined`, would you say it's acceptable to return `undefined` from `checkTimestampsValidityError`?